app/pages/BuyNBTC/useNBTC.tsx

```diff
@@ -14,13 +14,56 @@ import { logger } from "~/lib/log";
 const buyNBTCFunction = "buy_nbtc";
 const sellNBTCFunction = "sell_nbtc";
 
-async function getNBTCCoins(owner: string, client: SuiClient, nbtcCoin: string): Promise<PaginatedCoins> {
+async function getNbtcCoins(
+	owner: string,
+	client: SuiClient,
+	nbtcCoin: string,
+	cursor?: string | null,
+): Promise<PaginatedCoins> {
 	return client.getCoins({
 		owner,
 		coinType: nbtcCoin,
+		cursor,
 	});
 }
 
+// Returns the enough  nBTC coin array that has balance >= required amount
+export async function getEnoughNbtcCoinsWithAmount(
```

Comment:
1. this Thould be a generic function - should work with any coin type.
2. I'm 60% sure there is already a method to query the chain or graphql to find coins for a given amount.